### ISSUE TITLE
awaitFor does not fail on timeout for certain matchers

### ISSUE THREAD

**Issue by ssanj** [etorreborre/specs2#747]
This is a bit of a weird one but I can reproduce it consistently.

I'm seeing an issue when I use the `beGreaterThanOrEqualTo` matcher with a Future and an `awaitFor` that has a timeout. If the Future timesout, the test passes. With another matcher like `beEqualTo` the test fails when the Future times out. I expect the test to fail if the Future timesout.

I have added a sample test below:

```
package com.example.validation.extra

import org.specs2.mutable.Specification
import org.specs2.concurrent.ExecutionEnv
import scala.concurrent.Future
import scala.concurrent.duration._

final class PossibleBugSpec(implicit ee: ExecutionEnv) extends Specification {

  private val sleepyFuture: () => Future[Int] = () => 
    Future {
      Thread.sleep(2000)
      10
    }

  "Possible bug with Future and matchers" should {
    "with beEqualTo matcher" in {
      "pass on success" in {
        sleepyFuture() must beEqualTo(10).awaitFor(3.seconds)
      }

      "fail on timeout" in {
        sleepyFuture() must beEqualTo(10).awaitFor(1.second) must throwA[Exception]
      }
    }

    "with greaterThanOrEqualTo matcher" in {
      "pass on success" in {
        sleepyFuture() must beGreaterThanOrEqualTo(10).awaitFor(3.seconds)
      }

      "fail on timeout" in {
        //this should fail but doesn't.
        sleepyFuture() must beGreaterThanOrEqualTo(10).awaitFor(1.second) must throwA[Exception]
      }
    }
  }
}
```

The output from running the test is:

```
[info] PossibleBugSpec
[info] Possible bug with Future and matchers should
[info]   with beEqualTo matcher
[info]     + pass on success
[info]     + fail on timeout
[info]   with greaterThanOrEqualTo matcher
[info]     + pass on success
[error]     x fail on timeout
[error]      Expected: java.lang.Exception. Got nothing (PossibleBugSpec.scala:33)
[info] Total for specification PossibleBugSpec
```

I'm using `scala 2.12.8` with `specs2-core 4.5.1`

**Comment by etorreborre** [comment]
Hi Sanjiv, thanks for the report with a fully working example! I fixed this and am currently publishing a fix (`4.5.1-924020314`) in case you need it.

**Comment by ssanj** [comment]
Wow! Thanks for fixing this so quickly! 👍
